proposals/oras.md

```diff
@@ -0,0 +1,195 @@
+# OCI ORAS Project Proposal #
+
+## Abstract ##
+
+The [ORAS project][oras-project] is a CLI that can publish arbitrary content to an [OCI Spec Compliant](https://github.com/opencontainers/oci-conformance) registry, with special features for setting mediatypes on manifest configs and on content. 
+
+In order to provide OCI end users a method to publish and retrieve any type of content to/from an OCI registry, as well as a reference implementation for the artifacts project, ORAS should be moved under the opencontainers GitHub org.
+
+[oras-project]: https://github.com/deislabs/oras
+
+### ORAS Details ###
+
+ORAS is a CLI that can publish arbitrary content to an OCI registry, with special features for setting mediatypes on manifest configs and on content.
+
+Note: the manifest mediatype itself is always `application/vnd.oci.image.manifest.v1+json`.
+
+Example - uploading rockets, a brand new type of package:
+
+```
+# Create a thing
+printf '🚀' > rocket.txt
+
+# Create a manifest config
+printf '{"RocketVersion":"v0.1.0"}' > rocket-config.json
+
+# Upload your thing with a custom mediatype
+oras push localhost:5000/mystuff/myrocket:v0.1.0 rocket.txt:text/plain \
+  --manifest-config rocket-config.json:application/vnd.acme.rocket.config.v1+json
+ ```
+
+See manifest created:
+
+```
+$ curl -s -H 'Accept: application/vnd.oci.image.manifest.v1+json' \
+    http://localhost:5000/v2/mystuff/myrocket/manifests/v0.1.0 | jq
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.acme.rocket.config.v1+json",
+    "digest": "sha256:310175f34d2d4d5cba3418be06ddd1ef948147d729516d78318ec7f5c2d83d49",
+    "size": 26
+  },
+  "layers": [
+    {
+      "mediaType": "text/plain",
+      "digest": "sha256:ebbc0b2870eb323f2b6cffa5c493ceef81ae7eb36afc73d4e0367301631daec5",
+      "size": 4,
+      "annotations": {
+        "org.opencontainers.image.title": "rocket.txt"
```

Comment:
This is the default. It is basically the way oras knows how to write files to the current directory on pull.

However, this can be disabled by using custom layers/descriptors and `oras.WithNameValidation(nil)` on push.

The annotation itself was take from image-spec, see [here](https://github.com/opencontainers/image-spec/blob/da296dcb1e473a9b4e2d148941d7faa9ac8fea3f/annotations.md).

I wouldn't consider this a part of any spec surrounding artifacts, its just the way oras CLI chooses to store filenames. In the case of Helm, for example, it is entirely unused despite using oras as a library.

Reply:
The annotations are something we've debated. I'm not a big fan of them, and these should likely be something that went into the config for a specific artifact type. That said, it's interesting to see what's possible.